docs/migration-guides/qiskit-2.0.mdx

```diff
@@ -0,0 +1,260 @@
+This was strictly less featureful than the full control-flow form, and was never well-supported on hardware.
+This legacy method has been removed in Qiskit 2.0; you should use the [`if_test`](/api/qiskit/qiskit.circuit.QuantumCircuit#qiskit.circuit.QuantumCircuit.if_test) method with a proper condition.
+You can continue to use the 2-tuple form of `(register, integer)` to represent a register equality, or you can use the more expressive [classical computation subsystem](/api/qiskit/qiskit.circuit_classical) to represent more complex conditions.
+
+For example, you might have written:
+
+```python
+from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+qr = QuantumRegister(2, "q")
+cr = ClassicalRegister(2, "c")
+qc = QuantumCircuit(qr, cr)
+qc.h(0)
+qc.h(1)
+qc.measure(qr, cr)
+
+# Apply an X to qubit 0 if both measurements were 1.
+qc.x(0).c_if(cr, 3)
+```
+
+The most direct translation of this is to use the 2-tuple form with the [`if_test`](/api/qiskit/qiskit.circuit.QuantumCircuit#qiskit.circuit.QuantumCircuit.if_test) context manager:
+
+```python
+from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+qr = QuantumRegister(2, "q")
+cr = ClassicalRegister(2, "c")
+qc = QuantumCircuit(qr, cr)
+qc.h(0)
+qc.h(1)
+qc.measure(qr, cr)
+
+# Apply an X to qubit 0 if both measurements were 1.
+with qc.if_test((cr, 3)):
+    qc.x(0)
+```
+
+Note that in this form, you can also have the conditional gate access to multiple operations (simply add more into the indented block) without re-evaluating the conditions.
+You can also provide an ``else`` statement, and use more complex conditional logic:
+
+```python
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.classical import expr
+
+qc = QuantumCircuit(2, 2)
+qc.h(0)
+qc.h(1)
+qc.measure(qc.qubits, qc.clbits)
+
+# Apply an X to qubit 0 if both measurements were the same:
+with qc.if_test(expr.equal(qc.clbits[0], qc.clbits[1])) as else_:
+    qc.x(0)
+# ... or apply a Y to both qubits otherwise:
+with else_:
+    qc.y(0)
+    qc.y(1)
+```
+
+
+<span id="instruction_durations"></span>
+### Removal of `Instruction.duration` and `Instruction.unit`
+
+Circuit [`Instruction`](/api/qiskit/qiskit.circuit.Instruction) objects used to have attributes called `duration` and `unit`.
+These no longer exist in Qiskit 2.0.
+This is because the field generally only duplicated information that was a fixed property of hardware, and was stored explicitly in the [`Target`](/api/qiskit/qiskit.transpiler.Target).
+Storing the information separately for each individual instruction significantly increased the memory demands of Qiskit and slowed down even circuit operations that did not directly need to touch the information.
+
+The [`Delay`](/api/qiskit/qiskit.circuit#delay) instruction still contains these fields, as does the new-in-2.0 [`BoxOp`](/api/qiskit/qiskit.circuit#box), because these instructions have a _variable_ delay.
```

Comment:
```suggestion
The [`Delay`](/api/qiskit/qiskit.circuit#delay) instruction still contains these fields, as does the new-in-v2.0 [`BoxOp`](/api/qiskit/qiskit.circuit#box), because these instructions have a _variable_ delay.
```